### ISSUE TITLE
Wuhan issues review

### ISSUE THREAD

**Issue by anastassiavybornova** [uscuni/simplification#141]
**singles**

***

![image](https://github.com/user-attachments/assets/e709add2-1447-42ca-bcdd-9d8b89d8ae3b)
![image](https://github.com/user-attachments/assets/ae010136-c48c-4e5a-b135-e45827a6b99b)

adding these just to give an example - there's quite a lot of "almost overlapping" carriageways like that mapped out in wuhan (my suspicion is also that those are the ones that made it necessary to increase the eps)

***

![image](https://github.com/user-attachments/assets/e345ef31-55d6-4699-b260-c3097fbd983e)

simplification didn't change that one (cf for Liège singles!) - but could we just blame that on notsogreat mapping..?

***

![image](https://github.com/user-attachments/assets/ee92edbd-78df-422e-83b4-0eed87a85e1d)
![image](https://github.com/user-attachments/assets/9ba05464-6307-473f-8a00-4298f44e8be4)

same here, nothing changed - is that ok?

***

**Comment by anastassiavybornova** [comment]
**doubles**

***

![image](https://github.com/user-attachments/assets/cfad4c0c-b31b-4c68-b793-4d3c1af92edb)

why does only one of them get a dangle? ...or is that ok?

***

![image](https://github.com/user-attachments/assets/24a75f30-6ed5-4e3d-916d-5ff686df1b77)

potentially not what we wanted (dual carriageway type artifact that then becomes "twice as long")

***

![image](https://github.com/user-attachments/assets/dc5dfbab-4508-4d69-ab37-ea9f023ce6ec)


doubles where simplification didn't effect any change

***

![image](https://github.com/user-attachments/assets/991c8e0d-3980-4471-a017-186275b66284)

extreme edge case of "dangle would be needed"?

***

![image](https://github.com/user-attachments/assets/a1e84bfc-09c8-4fee-b9af-ba126f2dcd2b)


tricky one due to the mapping, non-planarity etc.

**Comment by anastassiavybornova** [comment]
**clusters (all except the giant one, cluster comp2)**

***

![image](https://github.com/user-attachments/assets/c8e4267c-cf3c-42d2-a18d-48fcd0075d09)


additional artifact created - but i suppose this one will be easy to handle with round2 of simplifications

***

![image](https://github.com/user-attachments/assets/e88d79d5-fb22-48c9-9f26-b206c6544992)
![image](https://github.com/user-attachments/assets/fcceeee8-4509-4387-a717-78a25a5ea0d6)


missing dangle? (example case for #117 )

***

![image](https://github.com/user-attachments/assets/c38ece10-5957-4369-a7ba-93f96891a9db)


did something get disconnected here or is it just looking like that on the plot?

***

**Comment by martinfleis** [comment]
![Screenshot 2024-08-05 at 12 03 08](https://github.com/user-attachments/assets/7d2c0d9b-7c51-4249-a2ed-00911e2ec5c8)

Also adding this missing connection in clusters

**Comment by anastassiavybornova** [comment]
# clusters: comp 2

this still doesn't work:

![image](https://github.com/user-attachments/assets/07eacbe4-be05-4994-afb1-bb4c533afccc)


***

trying to reproduce the skeletonization step by step for only comp2. could it be (in part?) because of these (places where the boundary goes through the polygon itself somehow, or where the polygon is only 1 node "thick"): 

![image](https://github.com/user-attachments/assets/2e863609-ee56-4ce3-a1e9-498f1a99cd9d)

![image](https://github.com/user-attachments/assets/9dc1ac19-1263-4ab9-bc80-513a1d415531)

**Comment by martinfleis** [comment]
That component worked on my side okay. Can you use other epsilon in the buffer?

**Comment by martinfleis** [comment]
> additional artifact created - but i suppose this one will be easy to handle with round2 of simplifications

I don't think so, the intersection on bottom left is non-planar.

**Comment by martinfleis** [comment]
> why does only one of them get a dangle? ...or is that ok?

because of the iterative treatment. The first one is considered a loop, hence we try a dangle. The other one is 3C'S where we don't try a dangle. I can try to include the dangle solution but I am afraid about the implications elsewhere.

**Comment by martinfleis** [comment]
> potentially not what we wanted (dual carriageway type artifact that then becomes "twice as long")

Yeah, that asks for a cluster solution but what would be the rule?

**Comment by anastassiavybornova** [comment]
> I can try to include the dangle solution but I am afraid about the implications elsewhere.

yeah, i see - i think then let's leave it like it is now!! this is really a minor detail

**Comment by anastassiavybornova** [comment]
> > potentially not what we wanted (dual carriageway type artifact that then becomes "twice as long")
> 
> Yeah, that asks for a cluster solution but what would be the rule?

i'm thinking that in a future iteration of the entire simplification process we could have some additional shape index based detection of artifacts caused specifically by dual carriageways, and specific rules for them? i would keep this in mind as a feature request rather than something urgent to implement - in the overwhelming majority of cases (i reviewed all of wuhan and all of liege) the dual carriageway artifacts got simplified just fine.

will check the wuhan pipeline with the larger eps now

**Comment by anastassiavybornova** [comment]
Current state of Wuhan issues: 

* introduce a "should we have a dangle here" check every time we drop S as a solution (instead of just being done with it)
* Doubles comp 39: xref #138, doubles comp 65)
* Clusters comp 2: solved now (with larger eps)
* Weird non-planarity cases: not solved - but also not possible to solve with what we're doing here (problem is in the data; we assume topological correctness for our input)
